src/Handlers/GeneralUiStateHandler.php

```diff
@@ -305,4 +320,48 @@ protected function _prepareLinksConfig($linksConfig)
 
         return $preparedLinksConfig;
     }
+
+    /**
+     * Convert an iterator to an array.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|Traversable $iterator The array or Traversable object to convert
+     *
+     * @throws InvalidArgumentException if $iterator is not an array or a Traversable object
+     *
+     * @return array Result of iterator to array transform.
+     */
+    protected function _iteratorToArrayRecursive($iterator)
+    {
+        if (!($iterator instanceof stdClass) && !is_array($iterator) && !$iterator instanceof Traversable) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an array or stdClass or Traversable object'), null, null, $iterator
+            );
+        }
+        if ($iterator instanceof stdClass) {
+            return (array) $iterator;
+        }
+        if (method_exists($iterator, 'toArray')) {
+            return $iterator->toArray();
+        }
+        $array = [];
+        foreach ($iterator as $key => $value) {
```

Comment:
Very un-normalized procedure.

1. Doing the same thing when scalar, and when is not `Traversable` or `array`.
2. Doing the same thing when `Traversable`, and when `array`.